Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -685,6 +682,30 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
         }
     }
 
+    private static func buildPaymentQueueWrapper(
+        systemInfo: SystemInfo,
+        operationDispatcher: OperationDispatcher
+    ) -> EitherPaymentQueueWrapper {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *),
```

Comment:
If we like this approach, we should extract this logic out to a common location, perhaps in `StoreKitVersion`